crates/apollo-parser/src/parser/mod.rs

```diff
@@ -157,10 +147,7 @@ impl Parser {
 
     /// Consume a token from the lexer and add it to the AST.
     fn eat(&mut self, kind: SyntaxKind) {
-        let token = self
-            .tokens
-            .pop()
-            .expect("Could not eat a token from the AST");
```

Comment:
I kept that possible panic, but is it really ok? There might be case where it is triggered. I'll test that through fuzzing